ipfs-cluster-ctl/main.go

```diff
@@ -151,6 +151,38 @@ requires authorization. implies --https, which you can disable with --force-http
 	}
 
 	app.Commands = []cli.Command{
+		{
+			Name:  "add",
+			Usage: "ipfs-cluster-ctl add <path> ... add a file to ipfs via cluster",
+			Description: `
+Only works with file paths, no directories.  Recurisive adding not yet supported.  --shard flag not
+yet supported.  Eventually users would use this endpoint if they want the file to be sharded across the cluster.
+This is useful in the case several ipfs peers want to ingest the file and combined have enough space
+to host but no single peer's repo has the capacity for the entire file.  No stdin reading yet either, that
+is also TODO
+`,
+			Flags: []cli.Flag{
+				cli.BoolFlag{
+					Name:  "recursive, r",
+					Usage: "add directory paths recursively, default false",
+				},
+			},
+			Action: func(c *cli.Context) error {
+				paths := make([]string, c.NArg(), c.NArg())
+				for i, path := range c.Args() {
+					paths[i] = path
+				}
+				// Unclear if multiFileR is ready for streaming, but hypothesis is yes.
+				// Files are all opened but not read until they are sent.
```

Comment:
Haha I had the same question... it seems so.. since `SerialFile` opens as `ReaderFile`

Reply:
In case you were wondering I ended up confident that cmdkit.multiFileRs are compatible with streaming huge inputs.  On a similar note cmdkit.multiPartFiles are also compatible which isn't super intuitive when you first look at the module.  As far as I can tell multiPartFiles break expected interface functionality to stay streamable.  When you call `NextFile` on a streamed directory that was somewhere in the middle of a file it always returns nil, the only directory that gets out files from calls to `NextFile` is the root directory (the first part over the wire).  To stream we can keep a reference to the rootfile and keep calling `NextFile` on it to generate each file in succession.

Reply:
yep